internals/src/Tizen.Applications.ThemeManager/Interop/Interop.ThemeManager.cs

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Runtime.InteropServices;
+using Tizen;
+using Tizen.Applications;
+
+internal static partial class Interop
+{
+    internal static partial class ThemeManager
+    {
+        private const string LogTag = "Tizen.Applications.ThemeManager";
+
+        internal enum ErrorCode : int
+        {
+            None = Tizen.Internals.Errors.ErrorCode.None,
+            InvalidParameter = Tizen.Internals.Errors.ErrorCode.InvalidParameter,
+            OutOfMemory = Tizen.Internals.Errors.ErrorCode.OutOfMemory,
+            PermissionDenied = Tizen.Internals.Errors.ErrorCode.PermissionDenied,
+            IoError = Tizen.Internals.Errors.ErrorCode.IoError,
+        }
+
+        internal static class ThemeManagerErrorFactory
+        {
+            internal static Exception GetException(Interop.ThemeManager.ErrorCode err, string message)
+            {
+                string errMessage = string.Format("{0} err = {1}", message, err);
+                Log.Warn(LogTag, errMessage);
+                switch (err)
+                {
+                    case Interop.ThemeManager.ErrorCode.InvalidParameter:
+                        return new ArgumentException(errMessage);
+                    case Interop.ThemeManager.ErrorCode.OutOfMemory:
+                        return new OutOfMemoryException(errMessage);
```

Comment:
We do not throw OutOfMemoryException on F/W code

❌ DO NOT explicitly throw OutOfMemoryException. This exception is to be thrown only by the CLR infrastructure.
https://docs.microsoft.com/ko-kr/dotnet/standard/design-guidelines/using-standard-exception-types#outofmemoryexception

Reply:
Done